mailpoet/lib/WooCommerce/Subscription.php

```diff
@@ -229,60 +237,64 @@ public function handleSubscriberOptin(SubscriberEntity $subscriberEntity, bool $
       return false;
     }
 
-    $subscriber->source = Source::WOOCOMMERCE_CHECKOUT;
+    $subscriber->setSource(Source::WOOCOMMERCE_CHECKOUT);
 
     if (
-      ($subscriber->status === SubscriberEntity::STATUS_SUBSCRIBED)
+      ($subscriber->getStatus() === SubscriberEntity::STATUS_SUBSCRIBED)
       || ((bool)$signupConfirmation['enabled'] === false)
     ) {
       $this->subscribe($subscriber);
     } else {
       $this->requireSubscriptionConfirmation($subscriber);
     }
 
-    SubscriberSegment::subscribeToSegments(
-      $subscriber,
-      array_merge([$wcSegment->id], $moreSegmentsToSubscribe)
-    );
+    $this->subscriberSegmentRepository->subscribeToSegments($subscriber, array_merge([$wcSegment], $moreSegmentsToSubscribe));
 
     return true;
   }
 
-  private function subscribe(Subscriber $subscriber) {
-    $subscriber->status = SubscriberEntity::STATUS_SUBSCRIBED;
-    if (empty($subscriber->confirmedIp) && empty($subscriber->confirmedAt)) {
-      $subscriber->confirmedIp = Helpers::getIP();
-      $subscriber->setExpr('confirmed_at', 'NOW()');
+  private function subscribe(SubscriberEntity $subscriber) {
+    $subscriber->setStatus(SubscriberEntity::STATUS_SUBSCRIBED);
+    if (empty($subscriber->getConfirmedIp()) && empty($subscriber->getConfirmedAt())) {
+      $subscriber->setConfirmedIp(Helpers::getIP());
+      $subscriber->setConfirmedAt(new Carbon());
```

Comment:
I think that we should use here a time from WP. Something like `Carbon::createFromTimestamp($this->wp->currentTime('timestamp'))`. Do you agree?

Reply:
Could you please elaborate on why you think we should use current_time() in this case? Is it because in this case we should respect whatever timezone is configured in WP? I haven't thought about it, it would be a change in the current behavior, but not necessarily a problem since what I'm suggesting is already changing the behavior anyway as the original code used MySQL's NOW().

Reply:
> Could you please elaborate on why you think we should use current_time() in this case?

What I know is that we use `current_time()` when we want to store current time. If I'm not wrong, the reason is to respect the user's timezone. Here is [an example](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/lib/Subscription/Pages.php#L190). But I guess it's not critical here.

Reply:
Yeah, I don't think it is critical and it would change the current behavior which doesn't take into consideration the user timezone set on WP. So I'm inclined to think we don't need to modify it in this PR. Let me know if you agree or not.